### ISSUE TITLE
RIR flexibility and lengthened partials implementation

### ISSUE THREAD

**Issue by WhyAsh5114** [WhyAsh5114/MyFit#160]
## Is your feature request related to a problem? Please describe.
#156

**Comment by Yun5Pc** [comment]
awesome the bug is fixed now 
i do have to say when you use the rename function it dosent delete the entry or take you off the page and the previous entry is still there and you can load it
but it just infinitely loads saying fetching performances until you reload the app
![image](https://github.com/user-attachments/assets/68f6cdaa-1340-43f3-ba9f-648624256036)

it might be nice to maybe get rid of that like perhaps adding a notice saying to reload or automatically reload the app
or idk if you can,maybe somehow delete it without having to reload the app

**Comment by Yun5Pc** [comment]
another issue ive just stumbled upon is this
![image](https://github.com/user-attachments/assets/dc5ab7b6-55e6-4611-9dcf-113cfa574047)
![image](https://github.com/user-attachments/assets/cb8df372-6f56-4978-8a7e-284fd0eba08b)
im guessing i used 0 weight to indicate bodyweight deadlifts back in v2 of the app
somehow using 0 load causes the relative overload chart to go nuts 
what do you think about this

**Comment by WhyAsh5114** [comment]
Yeah, nothing I can do for that tbh. I am planning to add a filter to the exercises where you can select the date from where u want to include them.

**Comment by WhyAsh5114** [comment]
I'll add a force reload for the infinite loading

**Comment by Yun5Pc** [comment]
> I'll add a force reload for the infinite loading

Awesome

> Yeah, nothing I can do for that tbh. I am planning to add a filter to the exercises where you can select the date from where u want to include them.

I was thinking that might be a good idea as well and maybe adding an option to filter by mesocycles might be beneficial as well

**Comment by Yun5Pc** [comment]
> Yeah, nothing I can do for that tbh. I am planning to add a filter to the exercises where you can select the date from where u want to include them.

As for the relative overload thing i realized that i had put 0 load and 0 reps to indicate that i had skipped those exercises that day
Since in v3 you can easily skip sets i dont think anyone else will run into that issue so thats really nice

Do you think that maybe making it so that you cant put 0 as reps or load would be better for new users and possibly stop the app from breaking in the future or just leaving as it is 
I wanted to ask this because you have already implemented this in the rir section where you get an error unless its greater or equal to 0

And just a quick question while we are talking about rir is there a specific reason why you have removed negative rir
I feel like negative rir could be used as a way to implement a system for recording maybe partial or forced reps beyond failure 

Would really like to know your perspective abouts all this

**Comment by WhyAsh5114** [comment]
For load there is a requirement to put at least 0.25 in case of non-bodyweight exercises, in bodyweight exercises you can have any load, even negative. I forgot to restrict the reps range to minimum 1, will add that in the next patch.

As for the RIR, negative RIRs would only make sense when you disable the `Force RIR matching` option in the mesocycle. I guess that's a valid approach as well, will add that too.

**Comment by Yun5Pc** [comment]
> For load there is a requirement to put at least 0.25 in case of non-bodyweight exercises, in bodyweight exercises you can have any load, even negative. I forgot to restrict the reps range to minimum 1, will add that in the next patch.

i guess this was implemented in v3 i never realized that was my bad

> As for the RIR, negative RIRs would only make sense when you disable the `Force RIR matching` option in the mesocycle. I guess that's a valid approach as well, will add that too.

hmm i dont seem to get why force rir matching needs to be disabled
idk whats best but

if we suppose that we did 12 reps and 3 partials we could put that as 15 reps with x load and -3 rir
or we could say that its 12 reps with x load -3 rir indicating you failed at 12 reps and did 3 partials
if we look at the former i dont think force rir would become a problem 
but it does get confusing a bit 
i dont usually do partials so i dont know the best way to actually journal them
what are your thoughts about this would it be better to add a seperate way to log partials

edit:even if we used the latter option for force rir matching i think we can use 12 as 0 rir and calculate that instead of calculating reps and load for -3rir

**Comment by WhyAsh5114** [comment]
RIR matching basically adjusts your reps to match the reps to allow progressive overload without leaving too many reps in the tank. **Force** RIR matching will decrease your reps to match the planned RIR of that microcycle (or week).

If we use -3 RIR on a set, and force RIR matching is enabled, the app will reduce the reps massively (at least 3) in order to **match** the planned RIR in the next microcycle. If you disable it, it'll work as you want it to, the app won't reduce the reps but it will attempt to reduce the RIR which is still not ideal tbh.

For LPs, I guess a completely different approach is needed. Its not something I'm prioritizing rn, I'm currently working on completely offline functionality of the app which is gonna take a while. I don't want to change the major workings of the app to avoid breaking something important.

**Comment by Yun5Pc** [comment]
> If we use -3 RIR on a set, and force RIR matching is enabled, the app will reduce the reps massively (at least 3) in order to **match** the planned RIR in the next microcycle. If you disable it, it'll work as you want it to, the app won't reduce the reps but it will attempt to reduce the RIR which is still not ideal tbh.

yeah so correct me if im wrong
you are basically saying if we did 12 reps at -3 rir when the planned rir for this microcycle is 0
in the next microcycle the app would give 9 reps at 0 right?
and if force match rir wasnt on the app would think it should be 12 reps at 0 rir is that right
while we are talking about this i really like the sliders where you allocate rir to microcycle when u are creating a meso
but i think it would be beneficial to add a way to perhaps remove specifc rirs for example
creating a meso cycle from 4-1rir removing the 0
or creating a meso cycle that starts with 3 rir skips 2 and goes to 1 rir and then 0
i think doing that will give users more flexibility
i know it probably isnt the most optimal way to train for people but i do think some people like to train at a relatively high rir like 3-2 rir for long periods because its sustainable and they can be consistent with it 
adding this flexibility would probably make the app attractive to more user groups i think

edit:a lil more feedback, for what ive said above since you already have this ui made
![image](https://github.com/user-attachments/assets/e20426e4-e47b-4fd0-8cb5-b12cc456d900)
ticking what rirs you want in this meso would be a really clean way to add that feature dont you think


> For LPs, I guess a completely different approach is needed. Its not something I'm prioritizing rn, I'm currently working on completely offline functionality of the app which is gonna take a while. I don't want to change the major workings of the app to avoid breaking something important.

that seems to be the better option i guess i dont think anyone is in need of this feature yet so dont worry about it bud im just giving some feedback as to how we can improve the app
its come a long way which is very very nice really do appreciate the hardwork your putting in 
prioritizing the offline mode is prolly major milestone id argue it would open up the app for more user groups im sure

as ive said i dont wanna be a bother lol if you ever have any features or updates to the app id be happy to help out by giving feedback as a user

edit: i dont know if ur exams are finished or not but again good luck with your studying bud i dont think anyone has any problems with the app atm so you should prolly take it easy and study if u have exams

**Comment by WhyAsh5114** [comment]
Ah no, exams are done. The feedback is definitely valuable and will keep this issue open for future reference, and will get back to it after the offline feature completes.

**Comment by Yun5Pc** [comment]
> Ah no, exams are done. The feedback is definitely valuable and will keep this issue open for future reference, and will get back to it after the offline feature completes.

awesome great to hear that

**Comment by Yun5Pc** [comment]
awesome work bud
